packages/assets-controllers/src/TokensController.ts

```diff
@@ -546,6 +558,24 @@ export class TokensController extends BaseController<
     return tokens[tokenIndex];
   }
 
+  /**
+   * This is a function that updates the tokens name for the tokens name if it is not defined.
+   *
+   * @param tokenList - Represents the fetched token list from service API
+   */
+  updateTokensName(tokenList: TokenListMap) {
```

Comment:
Just a suggestion, but, have you considered adding an `updateTokens` method instead? This would let us update token metadata including the name.

It just seemed a bit strange from an API design standpoint to have a function for updating a single property on each token, yet not any other properties.

Reply:
Thanks for the review @Gudahtt , I didn't considered that, something like this? 
` updateTokensAtribute(
    tokenList: TokenListMap,
    tokenAttribute: keyof Token & keyof TokenListToken,
  ) {
    const { tokens } = this.state;

    const newTokens = tokens.map((token) => {
      const newToken = tokenList[token.address.toLowerCase()];

      return !token[tokenAttribute] && newToken?.[tokenAttribute]
        ? { ...token, [tokenAttribute]: newToken[tokenAttribute] }
        : { ...token };
    });

    this.update({ tokens: newTokens });
  }
`

It's seems way better, let me know if you aggree

Reply:
Thanks @tommasini , that would make sense to me as well.

Though upon looking at this further, I'm having trouble grasping why the token detection controller is playing the role of updating token metadata here. I would expect the token detection controller to _just_ handle detection, not to synchronize metadata between the token list controller and the token controller. I see that this is a strange case though, sort of like a migration, since this `name` field was added after the fact. Hmm. Maybe I'll start a thread on Slack to discuss this further.

Reply:
Yes please, I'm indeed trying to cover the tokens that already existed and did not have the name property, since the UI of the [token list changed](https://recordit.co/H30qk9FcyH) (at least on mobile)